### PR TITLE
Expose Admin API to non-admin users

### PR DESCRIPTION
**Pull request by TheBitShepherd**
## Changes
* The `/admin/rest` API is now located under `/tenant/rest`, and can be optionally overriden with the `TENANT_AREA_PREFIX` environment variable.
* This API is available to tenant admins who have a valid access token for the tenant they are calling out to, and CRUD permissions for the models they are operating on. Session authentication is not permitted on these API endpoints.
* The django admin UI is still located under DJANGO_ADMIN_URL. Django-rq urls have also been moved under the DJANGO_ADMIN_URL prefix.
* Altered the multi-tenancy middleware to work with the new url prefix structure.

## Critical Changes
* Existing MetaDeploy services present in `cci` will no longer work. They will need to be re-connected using the new endpoint (`/tenant/rest`).

## QA Performed
* Registered a MetaDeploy service on cci using a token associated to tenant admin (non-super) user. Successfully able to run the `metadeploy_publish` task against EDA v1.120.

1. Logged in as a super user, ensure that a Site record exists with a domain of `localhost:8080`.
2. Create a "Tenant Admins" Permission group. Assign CRUD permissions for all models under the API application.
3. Login with devhub credentials and Django AllAuth will provision a new user.
4. In an incognito window, login as a super user, locate the new user that corresponds to your devhub user, and assign them the "Tenant Admins" group permissions.
5. Create a new token, and select the `localhost:8080` Site and devhub user for the lookups on the token.
6. Connect a new metadeploy service via `cci service connect metadeploy local`. For the domain enter `http://localhost:8080/tenant/admin`, copy and paste the token value from the django admin when prompted to enter it.
7. Clone the EDA repository and `cd` into it.
8. Run `cci task run metadeploy_publish --plan install --tag rel.1.120`
9. Verify that the EDA 1.120 Install plan was successfully uploaded via the django admin UI.

* Logged in as a tenant admin, and verified that `/tenant/rest/` endpoints are not accessible with a valid session.
* Verified that super users are still able to toggle between tenants via the picklist present in the Admin UI.
* The naming change from ADMIN_AREA_PREFIX to TENANT_AREA_PREFIX will not impact any of our apps as this environment variable is not currently in use.

### CODE REVIEW

File: metadeploy/api/schema.py [hunk 1]
```diff
@@ -2,10 +2,10 @@
 
 
 class MetaDeployEndpointEnumerator(EndpointEnumerator):
-    """Don't include the admin API in the schema and documentation"""
+    """Don't include the Tenant Admin API in the schema and documentation"""
 
     def should_include_endpoint(self, path, callback):
-        if path.startswith("/admin/rest/"):
+        if path.startswith("/tenant/rest/"):
```
Comment (TheBitShepherd) [inline metadeploy/api/schema.py:8]:
```suggestion
        if path.startswith("/{settings.TENANT_AREA_PREFIX}rest/"):
```

File: metadeploy/adminapi/tests/test_views.py [hunk 2]
```diff
@@ -1,61 +1,76 @@
...
+    perm = Permission.objects.get(
+        content_type=ContentType.objects.get_for_model(model_class),
+        codename=f"{perm}_{model_class._meta.model_name}",
+    )
+    user.user_permissions.add(perm)
+
 
 @pytest.fixture(name="assert_admin_api_multi_tenancy")
 def _multi_tenancy(assert_multi_tenancy):
     # Ensure the multi-tenancy client is a superuser, which is required by all these tests
```
Comment (davidmreed) [inline metadeploy/adminapi/tests/test_views.py:38]:
```suggestion
```

File: metadeploy/adminapi/tests/test_views.py [hunk 3]
```diff
@@ -437,39 +470,38 @@ def test_create_another_secondary(self, admin_api_client, plan_factory):
...
+    def test_ipaddress_restriction(self, token_client, plan_factory):
+        token_client.REMOTE_ADDR = "8.8.8.8"
 
         plan = plan_factory()
-        response = client.get(f"http://testserver/admin/rest/plans/{plan.id}")
+        add_perm_to_user(token_client.user, "Plan", Perm.VIEW)
+        response = token_client.get(f"{BASE_TENANT_API_URL}/plans/{plan.id}")
 
-        assert response.status_code == 400
+        assert response.status_code
```
Comment (davidmreed) [inline metadeploy/adminapi/tests/test_views.py:480]: What's the desired value for the status code here? 400/403? Can we be explicit?

Comment (TheBitShepherd) [reply metadeploy/adminapi/tests/test_views.py:480]: this was a goof, will correct. Thanks for the catch.

**Review by davidmreed — APPROVED** [review]

**Pull request merged**